tests/conftest.py

```diff
@@ -67,30 +84,93 @@ def docker_alluxio():
     subprocess.check_output(shlex.split(run_cmd_master))
     subprocess.check_output(shlex.split(run_cmd_worker))
     url = "http://127.0.0.1:28080"
-    timeout = 10
-    while True:
-        try:
-            LOGGER.debug("trying to connected to alluxio")
-            r = requests.get(url + "/v1/files?path=/")
-            LOGGER.debug("successfullly connected to alluxio")
-            if r.ok:
-                yield url
-                break
-        except Exception as e:  # noqa: E722
-            timeout -= 1
-            if timeout < 0:
-                raise SystemError from e
-            time.sleep(10)
+    yield yield_url(url)
     stop_docker(worker_container)
     stop_docker(master_container)
 
 
+@pytest.fixture(scope="module")
+def docker_alluxio_with_etcd():
+    if "ALLUXIO_URL" in os.environ:
```

Comment:
Can you merge the docker_alluxio() method and docker_alluxio_with_etcd() method to a method like launch_alluxio(etcd_hosts)
Most of the logics are shared